pkg/build/controller/build/build_controller.go

```diff
@@ -1872,6 +1893,39 @@ func (bc *BuildController) findOwnedConfigMap(owner *corev1.Pod, namespace strin
 	return true, nil
 }
 
+// taintedNamespace looks in the build namespace for pre existing ConfigMaps.
+//
+// These ConfigMaps should not exist if we are starting a new build. They should
+// be created only during the build boostrap process, not prior to it.
+func (bc *BuildController) taintedNamespace(build *buildv1.Build) (bool, error) {
+	for _, name := range []string{
+		buildutil.GetBuildCAConfigMapName(build),
+		buildutil.GetBuildGlobalCAConfigMapName(build),
+		buildutil.GetBuildSystemConfigMapName(build),
+	} {
+		if hasCM, err := bc.findConfigMap(
+			build.Namespace, name,
+		); err != nil {
+			return false, err
+		} else if hasCM {
+			return true, nil
```

Comment:
I think this can return a false positive. Consider the following scenario:

1. Build is created
2. Build controller creates the build pod
3. Build controller creates the additional CA configMap
4. Build controller gets an apiserver error creating the global trust CA configMap

@bparees based on my understanding, in this scenario the build controller will re-sync and try to continue setting up the build pod. With this PR the build will fail permanently because one of the ConfigMaps already exists.

Reply:
That is not the case @adambkaplan. If the pod already exists we don't check for "tainted namespace", we just operate as we used to do prior to this PR. Is there something I am not seeing?